### ISSUE TITLE
OSPF routes are not imported

### ISSUE THREAD

**Issue by netfreak98** [natesales/bcg#8]
Hi again :)

it's not possible to import OSPF routes due to the default generated kernel filter:

```
protocol kernel {
  scan time 10;
  ipv4 {
    export filter {
      if (proto = "explstatic4" || proto = "explstatic6") then accept;
      if source = RTS_STATIC && proto != "static4" then {
        accept;
      } else if source = RTS_BGP then {
        krt_prefsrc = 185.121.69.144;
        accept;
      }
      reject;

    };
  };
```

Would it be possible to add "protocols" in the global config please which will then add something like this:

```
cat config.yml

asn: 49697
router-id: 185.121.69.144
pref-src4: 185.121.69.144
pref-src6: 2001:678:a00:1000::1
allowed-proto-ipv4:
  - "ALLOWED_PROTO4_FROM_CONFIG1"
  - "ALLOWED_PROTO4_FROM_CONFIG2"
allowed-proto-ipv6:
  - "ALLOWED_PROTO6_FROM_CONFIG1"
  - "ALLOWED_PROTO6_FROM_CONFIG2"
 [..]
```

Which adds:
```
protocol kernel {
  scan time 10;
  ipv4 {
    export filter {
     [..]
      if (proto = "ALLOWED_PROTO4_FROM_CONFIG1") then accept;
      if (proto = "ALLOWED_PROTO4_FROM_CONFIG2") then accept;
    [..]

```
and

```
protocol kernel {
  scan time 10;
  ipv6 {
    export filter {
    [..]
      if (proto = "ALLOWED_PROTO6_FROM_CONFIG1") then accept;
      if (proto = "ALLOWED_PROTO6_FROM_CONFIG2") then accept;
    [..]
```

By the way.. why is "explstatic6" configured under the ipv4 address family channel below the first kernel statement?

```
protocol kernel {
  scan time 10;
  ipv4 {
    export filter {
      if (proto = "explstatic4" || proto = "explstatic6") then accept;
```

Shouldn't it be like below? or am I wrong? :D 

```
protocol kernel {
  scan time 10;
  ipv4 {
    export filter {
      if (proto = "explstatic4" then accept;
```
and
```
protocol kernel {
  scan time 10;
  ipv6 {
    export filter {
      if (proto = "explstatic6") then accept;

```

**Comment by netfreak98** [comment]
my workaround for now is using ospfd from FRR :D

**Comment by natesales** [comment]
Nice catch on the explicit static routes, fixed in 420c0c2db8c037e8ce0c1d6fe1cd36b3e507d39f

I've also added the ability to inject arbitrary protocols into the kernel, defined in a2c6c3540c7d0ee48cd6d045c20669168d1ff7cc. The config keys are `kernel-inject4` and `kernel-inject6`
